src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs

```diff
@@ -199,23 +199,26 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
             for (int iinfo = 0; iinfo < Bindings.InfoCount; ++iinfo)
                 outColumnNames[iinfo] = Bindings.GetColumnName(Bindings.MapIinfoToCol(iinfo));
 
-            string opType = "Binarizer";
+            string scoreColumn = Bindings.RowMapper.OutputSchema[Bindings.ScoreColumnIndex].Name;
+
             OnnxNode node;
-            var binarizerOutput = ctx.AddIntermediateVariable(null, "BinarizerOutput", true);
+            string opType = "Binarizer";
+            var binarizerOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "BinarizerOutput", false);
+            node = ctx.CreateNode(opType, ctx.GetVariableName(scoreColumn), binarizerOutput, ctx.GetNodeName(opType));
+            node.AddAttribute("threshold", _threshold);
 
-            string scoreColumn;
-            if (Bindings.RowMapper.OutputSchema[Bindings.ScoreColumnIndex].Name == "Score")
-                scoreColumn = outColumnNames[1];
-            else
+            string comparisonOutput = binarizerOutput;
+            if (Bindings.PredColType is KeyDataViewType)
             {
-                Host.Assert(Bindings.InfoCount >= 3);
-                scoreColumn = outColumnNames[2];
+                var one = ctx.AddInitializer(1.0f, "one");
```

Comment:
>ctx.AddInitializer(1.0f, "one"); [](start = 26, length = 32)

Is this +1 ? to be 1 based?

Reply:
Yes, this is to make it one based and make it consistent with ML.NET results.

---
In reply to: [383088629](https://github.com/dotnet/machinelearning/pull/4878#discussion_r383088629) [](ancestors = 383088629)